compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java

```diff
@@ -57,20 +61,65 @@
 public class BIRInstructionWriter extends BIRVisitor {
 
     private ByteBuf buf;
+    private ByteBuf scopeBuf;
     private ConstantPool cp;
     private BIRBinaryWriter binaryWriter;
+    private int instructionOffset;
+    private Set<BirScope> completedScopeSet;
+    private int scopeCount;
 
-    BIRInstructionWriter(ByteBuf buf, ConstantPool cp, BIRBinaryWriter birBinaryWriter) {
+    BIRInstructionWriter(ByteBuf buf, ByteBuf scopeBuf, ConstantPool cp, BIRBinaryWriter birBinaryWriter) {
         this.buf = buf;
+        this.scopeBuf = scopeBuf;
         this.binaryWriter = birBinaryWriter;
         this.cp = cp;
+        this.instructionOffset = 0;
+        this.completedScopeSet = new HashSet<>();
+        this.scopeCount = 0;
+    }
+
+    BIRInstructionWriter(ByteBuf buf, ConstantPool cp, BIRBinaryWriter birBinaryWriter) {
+        this(buf, null, cp, birBinaryWriter);
+    }
```

Comment:
we don't need this new constructor as discussed.

Reply:
Fixed